tests/test_userwallet.py

```diff
@@ -1,62 +1,105 @@
-import asyncio
+import pytest
 
 from aiohttp.client import ClientSession
 
 from pylnbits.config import Config
 from pylnbits.user_wallet import UserWallet
 
-# Example code
 """
 Tests: 
- decode an invoice 
- Get invoices (incoming or out) 
+ 
+ Get wallet details
+ Check invoice
+ Pay invoice
+ Pay lnurl
+ Pay lightning address
+ Create an invoice
+ Decode an invoice
+ Get payment hash
+ Get invoices (incoming or out)
  Get invoice(s) by memo (incoming or out)
 """
 
-async def main():
+class TestUserWallet:
 
-    c = Config(config_file="config.yml")
-    url = c.lnbits_url
-    print(f"url: {url}")
-    print(f"headers: {c.headers()}")
-    print(f"admin_headers: {c.admin_headers()}")
+    @pytest.fixture(autouse=True)
+    async def setup_vars(self):
+        c = Config(config_file="config.yml")
+        session = ClientSession()
+        self.uw = UserWallet(c, session)
+        yield
+        await session.close()
+    
+    # get wallet details
+    async def test_get_wallet_details(self):
+        userwallet = await self.uw.get_wallet_details()
+        assert userwallet is not None, "Failed to get wallet details"
+        print(f"\nuser wallet info : {userwallet}")
+    
+    # check invoice
+    async def test_check_invoice(self):
+        payment_hash = "edefef3766537446c70e51af9b414fb3b319baf515f1ff9852c0289eae3665a1"
+        res = await self.uw.check_invoice(payment_hash)
+        assert "paid" in res, res.get("detail", "Check failed")
+        print(f"\nInvoice status: {res['paid']}. \n\nFull check invoice response: \n{res}")
 
-    async with ClientSession() as session:
-        uw = UserWallet(c, session)
-        # works
-        userwallet = await uw.get_wallet_details()
-        print(f"user wallet info : {userwallet}")
+    # pay invoice
+    async def test_pay_invoice(self):
+        bolt = "lnbc800n1ps23r2dpp5ahh77dmx2d6yd3cw2xheks20kwe3nwh4zhcllxzjcq5fat3kvkssdqsd9h8vmmfvdjk7mn9cqzpgrzjq02snzwz4petaly54yzjkm358rqa5as9hkgydjvxxmvlpuk6dfd9cz0y2cqq0qsqqyqqqqlgqqqqqqgq9qsp5cut63ftfcffwkrr2w9r50w5e40m93k3er75mc70ysxps7yercs9s9qyyssqs7qk3cz97nm5m6ehzedcxhttx87l7x5kk38gvwkzzv4lhrhddtqq3sk43nnvsddagf36ledw9vhlpqxuu5s53pj6sz926mwqxf8chsgp2m9j8w"  
```

Comment:
not mandatory but might be good to an option to test these by asking for the bolt at runtime on cli when running the pytest

Reply:
This is a really good idea

Reply:
Implemented this in: https://github.com/lightningames/pylnbits/pull/178